src/ClientData/ScopeIdProvider.cpp

```diff
@@ -26,10 +30,20 @@ namespace orbit_client_data {
                 std::begin(instrumented_functions), std::end(instrumented_functions),
                 [](const auto& a, const auto& b) { return a.function_id() < b.function_id(); })
                 ->function_id();
-  return std::unique_ptr<NameEqualityScopeIdProvider>(new NameEqualityScopeIdProvider(max_id + 1));
+
+  absl::flat_hash_map<uint64_t, std::string> scope_id_to_name;
+  std::transform(std::begin(instrumented_functions), std::end(instrumented_functions),
+                 std::inserter(scope_id_to_name, std::end(scope_id_to_name)),
+                 [](const auto& instrumented_function) {
+                   return std::make_pair(instrumented_function.function_id(),
+                                         instrumented_function.function_name());
+                 });
```

Comment:
I know you like `std::transform` and I'm fine with it, but I want to note how a for loop here would:
- be shorter (3 lines vs 6) and arguably more readable (no `begin`, `end`, `inserter`, `make_pair`, lambda);
- skip the construction of the pair with `emplace`.

Reply:
I was thinking the same, didn't commented. But as we already 2, maybe its not worth having the transform.

Reply:
But... expressiveness...

Reply:
But in C++ this is just harder to read. In Scala (or similar), I would definitely prefer the `map` over any kind of loop. But in C++...

Reply:
Yeah, I mean, the only advantage of transform we still have here is expressiveness. Yet it is also hard to see behind the 6 lines. That comment was self-irony.